data-prepper-plugins/newline-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/newline/NewlineDelimitedOutputCodec.java

```diff
@@ -4,42 +4,89 @@
  */
 package org.opensearch.dataprepper.plugins.codec.newline;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
 import org.opensearch.dataprepper.model.codec.OutputCodec;
 import org.opensearch.dataprepper.model.event.Event;
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.util.Map;
+import java.util.Objects;
 
 /**
  * An implementation of {@link OutputCodec} which deserializes Data-Prepper events
  * and writes them to Output Stream as ND-JSON data
  */
-@DataPrepperPlugin(name = "newline", pluginType = OutputCodec.class)
+@DataPrepperPlugin(name = "newline", pluginType = OutputCodec.class, pluginConfigurationType = NewlineDelimitedOutputConfig.class)
 public class NewlineDelimitedOutputCodec implements OutputCodec {
+    private static final String NDJSON = "ndjson";
+    private static final String MESSAGE_FIELD_NAME = "message";
```

Comment:
This message key is normally just set in the sources for raw logs that are not json. How is it useful in the sink? Won't all Events already be in json format at this point, at which point `message` is irrelevant.

Reply:
we are using message key in input codec while putting data in Map
<img width="328" alt="image" src="https://github.com/opensearch-project/data-prepper/assets/133762828/571512ad-2c86-473a-b809-21c36fa5ce42">


we are doing reverse of that so that we can extract the data here in outputcodec